### ISSUE TITLE
Have TQDMProgressBar work in evaluate mode as well

### ISSUE THREAD

**Issue by zaccharieramzi** [tensorflow/addons#1620]
**Describe the feature and the current behavior/state.**

Right now, the `TQDMProgressBar` callback can only be used when training a model.
When evaluating nothing happens. And even if I try to "cheat" by manually setting the `on_test_batch_(begin|end)` methods, it fails with an error about `mode`.

**Relevant information**
- Are you willing to contribute it (yes/no): don't know
- Are you willing to maintain it going forward? (yes/no): no
- Is there a relevant academic paper? (if so, where): no
- Is there already an implementation in another framework? (if so, where): no
- Was it part of tf.contrib? (if so, where): no

**Which API type would this fall under (layer, metric, optimizer, etc.)**
callbacks

**Who will benefit with this feature?**
everyone using `tqdm` to evaluate in a notebook: I want something fancier than the default evaluate progressbar

**Any other info.**
It failed in both tf2.2 and tf2.1

You can find the colab gist [here](https://colab.research.google.com/drive/1eHI-6QhttMhdMYeH8VP0SwfyB0b46ugj).

**Comment by gabrieldemarmiesse** [comment]
I don't think it would be hard to do because keras exposes methods for that: e.g. https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback#on_test_batch_begin . Contributions welcome!

**Comment by ashutosh1919** [comment]
@gabrieldemarmiesse , please review PR [#38447](https://github.com/tensorflow/tensorflow/pull/38447)

**Comment by gabrieldemarmiesse** [comment]
As a small note, I believe the TQDM callback should have a cleanup as we removed the "samples" mode and a refactoring would be welcome before a PR to add this feature. There is definitly some dead code/renaming to do.

**Comment by ashutosh1919** [comment]
Yep. Got it. Let me do that too.

**Comment by ashutosh1919** [comment]
@gabrieldemarmiesse , I have created PR for adding methods `on_test_batch_begin()` and `on_test_batch_end()`. I have also looked at removing unused code related to `samples` mode. I have found one clue [here](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/tqdm_progress_bar.py#L161). But since, I am not aware about past implementation and changes, I couldn't do on my own. Please give me hints or point me the direction where I can find one.

**Comment by shun-lin** [comment]
ops, sorry I have forgot about evaluate() when I wrote it and thanks @ashutosh1919 so much for the contribution!

**Comment by ashutosh1919** [comment]
@zaccharieramzi and @gabrieldemarmiesse , please review PR. Creates as below.
![tqdm_pbar_demo - Colaboratory](https://user-images.githubusercontent.com/20843596/79098888-8b070380-7d80-11ea-9043-117eb406426a.gif)
